File: src/main/java/com/intetics/lukyanenko/models/AppUser.java
package com.intetics.lukyanenko.models;

import java.util.List;

public class AppUser extends Common
{
  private String name;
  private String password;
  private boolean isCustomer;
  private List<String> roles;
  private boolean isNew;
  
  public AppUser()
  {
    super();
    isNew = true;
  }
  
  public AppUser(boolean isNew)
  {
    this.isNew = isNew;
  }
  
  public AppUser(String name)
  {
    this(false);
    this.name = name;
  }
  
  public String getName() {
    return name;
  }
  
  public void setName(String name) {
    this.name = name;
  }
  
  public String getPassword() {
    return password;
  }
  
  public void setPassword(String password) {
    this.password = password;
  }
  
  public List<String> getRoles() {
    return roles;
  }
  
  public void setRoles(List<String> roles) {
    this.roles = roles;
  }
  
  public boolean getIsCustomer()
  {
    return isCustomer;
  }
  
  public void setIsCustomer(boolean customer)
  {
    isCustomer = customer;
  }
  
  public boolean getIsNew()
  {
    return isNew;
  }
  
  public void setIsNew(boolean isNew)
  {
    this.isNew = isNew;
  }
}

File: src/main/java/com/intetics/lukyanenko/models/AppUser.kt
package com.intetics.lukyanenko.models

class AppUser : Common {
    lateinit var name: String
    lateinit var password: String
    var isCustomer: Boolean = false
    var isNew: Boolean = false
    var roles: Set<String> = HashSet<String>()

    constructor() : super() {
        isNew = true
    }

    constructor(isNew: Boolean) {
        this.isNew = isNew
    }

    constructor(name: String) : this(false) {
        this.name = name
    }
}
